python-checks/src/main/java/org/sonar/python/checks/UnusedFunctionParameterCheck.java

```diff
@@ -53,21 +58,43 @@ private static void checkFunctionParameter(SubscriptionContext ctx, FunctionDef
       .forEach(param -> ctx.addIssue(param, String.format(MESSAGE, param.name().name())));
   }
 
+  private static boolean isException(FunctionDef functionDef) {
+    FunctionSymbol functionSymbol = ((FunctionDefImpl) functionDef).functionSymbol();
+    return CheckUtils.containsCallToLocalsFunction(functionDef) ||
+      SymbolUtils.canOverrideMethod(functionSymbol) ||
+      isInterfaceMethod(functionDef) ||
+      !functionDef.decorators().isEmpty() ||
+      isSpecialMethod(functionDef) ||
+      hasNonCallUsages(functionSymbol);
+  }
+
   private static boolean isInterfaceMethod(FunctionDef functionDef) {
     return functionDef.body().statements().stream()
       .allMatch(statement -> statement.is(Kind.PASS_STMT, Kind.RAISE_STMT)
-        || (statement.is(Kind.EXPRESSION_STMT) && isStringExpression((ExpressionStatement) statement)));
+        || (statement.is(Kind.EXPRESSION_STMT) && isStringExpressionOrEllipsis((ExpressionStatement) statement)));
+  }
+
+  private static boolean isStringExpressionOrEllipsis(ExpressionStatement stmt) {
+    return stmt.expressions().stream().allMatch(expr -> expr.is(Kind.STRING_LITERAL, Kind.ELLIPSIS));
+  }
+
+  private static boolean isSpecialMethod(FunctionDef functionDef) {
+    String name = functionDef.name().name();
+    return name.startsWith("__") && name.endsWith("__");
```

Comment:
In the ticket we also mention methods starting with a single `_`. Do we want to report on those? It might make sense but we should update the ticket. Currently we have a test asserting a private method is not raising but it seems to be due to an unresolved parent than due to the fact the method is "private" (starting with `_` at least).

Reply:
I think we shouldn't exclude methods starting with a single `_` , I will remove it from the ticket!